aws/table_aws_codestar_notification_rule.go

```diff
@@ -0,0 +1,193 @@
+package aws
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/codestarnotifications"
+
+	"github.com/turbot/go-kit/types"
+
+	"github.com/turbot/steampipe-plugin-sdk/v5/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsCodestarNotificationRule(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_codestar_notification_rule",
+		Description: "AWS CodeStar Notification Rule",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.SingleColumn("arn"),
+			IgnoreConfig: &plugin.IgnoreConfig{
+				ShouldIgnoreErrorFunc: shouldIgnoreErrors([]string{"NotFound", "InvalidParameter"}),
```

Comment:
Hi @gertvv, should we consider handling the `InvalidParameter` exception here? I believe it would be beneficial to inform the user that the provided value is invalid, allowing them to correct it. What are your thoughts on this?


Additionally, the API [DescribeNotificationRule](https://docs.aws.amazon.com/codestar-notifications/latest/APIReference/API_DescribeNotificationRule.html#API_DescribeNotificationRule_Errors) returns a `ResourceNotFoundException` instead of `NotFound`. Please update the not found error code accordingly.

Reply:
That sounds good, but after some fiddling I haven't been able provoke an InvalidParameter error, and I'm also not sure where I can handle it. Some pointers for both would be useful.

Reply:
I've removed the ignore for InvalidParameter. Using an invalid `where arn = 'abcdef'` clause causes a permissions error instead of this. And nonsense values for the filter parameters do not result in errors.